### PR TITLE
Junos Hostname Resource Module Implemented

### PR DESCRIPTION
**Pull request by rohitthakur2590**
Signed-off-by: Rohit Thakur <rohitthakur2590@outlook.com>

##### ISSUE TYPE

 
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/junos_hostname.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "network",
+}
+
+DOCUMENTATION = """
+---
+module: junos_hostname
+version_added: 2.9.0
+short_description: Manage SNMP server configuration on Junos devices.
```
Comment (GomathiselviS) [inline plugins/modules/junos_hostname.py:43]:
```suggestion
short_description: Manage hostname configuration on Junos devices.
```

File: plugins/modules/junos_hostname.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+description: This module manages SNMP server configuration on devices running Junos.
```
Comment (GomathiselviS) [inline plugins/modules/junos_hostname.py:44]:
```suggestion
description: This module manages hostname configuration on devices running Junos.
```

File: plugins/modules/junos_hostname.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+  config:
+    description: A dictionary of SNMP server configuration.
+    type: dict
+    suboptions:
+      hostname:
+        description: Specify the hostname.
+        type: str
+  state:
+    description:
+    - The state the configuration should be left in.
```
Comment (GomathiselviS) [inline plugins/modules/junos_hostname.py:73]: I think we should mention that the merged, replaced, and overridden states are going to behave the same for this module.

File: tests/integration/targets/junos_hostname/tests/netconf/_populate_config.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
```
Comment (GomathiselviS) [inline tests/integration/targets/junos_hostname/tests/netconf/_populate_config.yaml:1]: This file has to be removed.

**Review by GomathiselviS — CHANGES_REQUESTED** [review]
There are some minor changes to be done. Otherwise, the module looks good to be merged.

File: plugins/module_utils/network/junos/facts/hostname/hostname.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+          from spec for null values
+
+        :param spec: The facts tree, generated from the argspec
+        :param conf: The configuration
+        :rtype: dictionary
+        :returns: The generated config
+        """
+        hostname_config = {}
+
+        # Parse facts for BGP address-family global node
```
Comment (KB-perByte) [inline plugins/module_utils/network/junos/facts/hostname/hostname.py:131]: We can remove the comment  here

File: plugins/modules/junos_hostname.py [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+  returned: when changed
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+  type: dict
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: always
+  type: list
+  sample: ['"<nc:host-name>78.46.194.186</nc:host-name></nc:system>"', 'command 2', 'command 3']
```
Comment (KB-perByte) [inline plugins/modules/junos_hostname.py:323]: we may remove command 2 and 3 and keep the relevant one only

**Review by KB-perByte — APPROVED** [review]

**Review by GomathiselviS — APPROVED** [review]

**Pull request merged**